lib/vimgolf/challenge.rb

```diff
@@ -34,9 +55,14 @@ def download
         @data['in']['data'].gsub!(/\r\n/, "\n")
         @data['out']['data'].gsub!(/\r\n/, "\n")
 
-        @type = @data['in']['type']
+        # be sure to sanitize the types
+        @type = @data['in']['type'].gsub(/[^\w-]/, '.')
```

Comment:
Should we be using shellescape instead? http://www.ruby-doc.org/stdlib-1.9.3/libdoc/shellwords/rdoc/Shellwords.html#method-c-shellescape